### ISSUE TITLE
Adobe CF2018 Compatibility

### ISSUE THREAD

**Issue by bockensm** [cfchef/fw1-logbox-example#1]
In the readme you mention that administrator mappings would be required to get ColdFusion compatibility. The repo runs out of the box with Lucee, but if I set `cfengine=adobe@2018` I get the following error:

```
bean not found: LogBoxConfig
```

Do you have any more information on what mappings would be required for this to work with ColdFusion?

**Comment by tonyjunkes** [comment]
Hi @bockensm,

I set this example up with FW/1’s core files outside of the webroot. When extending components, Lucee has a nice feature where you can use a relative path as the attribute value instead of a dotted delimiter path.

Since ACF does not support this, you can add a mapping in the ColdFusion Admin that maps /framework to the install path of FW/1 and change the extends value to “framework.one”. That should fix any lookup issues, I think.

Honestly for the sake of making this a better all around example, I should update this to have everything in the webroot so it’s easier to run across engines. I can make this change soon.

Let me know if my suggestion is of any use. I don’t think I ever actually tested the example on ACF so there certainly could be another config issue I did not foresee.

Cheers.

**Comment by bockensm** [comment]
@tonyjunkes I pushed everything in to /app and took a more traditional approach to it and that's when I got the error I mentioned above. I sent a PR that should let you see the exact situation I set up and the error I received.

**Comment by tonyjunkes** [comment]
@bockensm I pushed changes that should resolve the ACF error.

These changes consolidated everything into the webroot similar to how you moved it all into /app and I also cleaned up the code a little - updated LogBox to a newer version, instructed DI/1 to create the LogBox instances as transients as I'm pretty sure that's what ColdBox would do (this was also a test to see if that was bottlenecking ACF).

ACF is apparently doing something different from Lucee when DI/1's builder DSL constructs bean declarations chained into one call. Lucee happily returns the updated bean factory instance with the newly added beans every time `done()` is called. ACF doesn't like this for reasons unknown to me at this time. The fix was to separate the bean declarations into 2 calls.

This could be a bug in DI/1. I'd lean towards saying whatever Lucee is doing is correct as it's expected; so DI/1 may need to be tweaked to cater to ACF's odd behavior. I'll dig into that some more but I couldn't see a way around it otherwise. I'm def gonna circle back to this some more.

Let me know how the changes pan out for you.
